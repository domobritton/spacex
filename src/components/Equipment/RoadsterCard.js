import React from 'react'
import styled from 'styled-components'

export const RoadsterCard = ({ roadster }) => {
    let utcDate = roadster.launch_date_utc
    let localDate = new Date(utcDate).toLocaleDateString()
    let distance = (Math.round(roadster.earth_distance_mi * 100) / 100).toString()
    let speed = (Math.round(roadster.speed_mph * 100) / 100).toString() 

    return (
        <>
            <List>
                <Item>Launch Date: {localDate}</Item>
                <Item>Distance From Earth: {distance} miles</Item>
                <Item>Travel Speed: {speed} miles per hour</Item>
                <Item>Orbit Type: {roadster.orbit_type}</Item>
            </List>
            <SubTitle>{roadster.name}</SubTitle>
            <Description>
                {roadster.details}
                <Images>
                    <TeslaImg src={roadster.flickr_images[0]} />
                    <TeslaImg src={roadster.flickr_images[1]} />
                    <TeslaImg src={roadster.flickr_images[2]} />
                    <TeslaImg src={roadster.flickr_images[3]} />
                </Images>
            </Description>
        </>
    )
}

const List = styled.ul `
    display: flex;
    position: relative;
    justify-content: space-around;
    align-items: center;
    background: #1C1F1F;
    height: 32px;

    @media all and (max-width: 768px) {
        margin: 40px;
    }
`;
List.displayName = 'List'

const Item = styled.li `
    list-style: none;
    color: #A7A8A8;
    font-size: 12px;
    font-weight: bold;
`;
Item.displayName = 'Item'

const SubTitle = styled.h3 `
    font-size: 30px;
    margin: 50px 20px 30px;
`;
SubTitle.displayName = 'Subtitle'

const Description = styled.div `
    position: relative;
    padding: 40px 20px;
    line-height: 24px;

    @media all and (max-width: 768px) {
        margin: 40px;
    }
`;
Description.displayName = 'Description'

const Images = styled.div `
    display: flex;
    justify-content: space-around;
    align-items: center;
    margin-top: 30px;
`;
Images.displayName = 'Images'

const TeslaImg = styled.img `
    width: 22.5%;
    overflow: hidden;
`;
TeslaImg.displayName = 'TeslaImg'