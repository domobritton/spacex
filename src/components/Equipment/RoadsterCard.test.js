import { RoadsterCard } from './RoadsterCard'

const data = {
    'name': `Elon Musk's Tesla Roadster`,
    'launch_date_utc': '2018-02-06T20:45:00.000Z',
    'orbit_type': 'heliocentric',
    'speed_mph': 45362.54860012799,
    'earth_distance_mi': 210452509.00200382,
    'flickr_images': [
        'https://farm5.staticflickr.com/4615/40143096241_11128929df_b.jpg',
        'https://farm5.staticflickr.com/4702/40110298232_91b32d0cc0_b.jpg',
        'https://farm5.staticflickr.com/4676/40110297852_5e794b3258_b.jpg',
        'https://farm5.staticflickr.com/4745/40110304192_6e3e9a7a1b_b.jpg'
    ],
    'details': `Elon Musk's Tesla Roadster`
}

it('renders correctly', () => {
    const wrapper = render(
        <RoadsterCard roadster={data}/>
    )

    expect(wrapper).toMatchSnapshot()
})
