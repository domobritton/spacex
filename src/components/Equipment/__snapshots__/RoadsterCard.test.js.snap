// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <ul
    class="sc-bdVaJa hmxnZd"
  >
    <li
      class="sc-bwzfXH eEHmLi"
    >
      Launch Date: 2/6/2018
    </li>
    <li
      class="sc-bwzfXH eEHmLi"
    >
      Distance From Earth: 210452509 miles
    </li>
    <li
      class="sc-bwzfXH eEHmLi"
    >
      Travel Speed: 45362.55 miles per hour
    </li>
    <li
      class="sc-bwzfXH eEHmLi"
    >
      Orbit Type: heliocentric
    </li>
  </ul>,
  <h3
    class="sc-htpNat lbrqdO"
  >
    Elon Musk's Tesla Roadster
  </h3>,
  <div
    class="sc-bxivhb CNtsX"
  >
    Elon Musk's Tesla Roadster
    <div
      class="sc-ifAKCX dRaxcw"
    >
      <img
        class="sc-EHOje ikBWqV"
        src="https://farm5.staticflickr.com/4615/40143096241_11128929df_b.jpg"
      />
      <img
        class="sc-EHOje ikBWqV"
        src="https://farm5.staticflickr.com/4702/40110298232_91b32d0cc0_b.jpg"
      />
      <img
        class="sc-EHOje ikBWqV"
        src="https://farm5.staticflickr.com/4676/40110297852_5e794b3258_b.jpg"
      />
      <img
        class="sc-EHOje ikBWqV"
        src="https://farm5.staticflickr.com/4745/40110304192_6e3e9a7a1b_b.jpg"
      />
    </div>
  </div>,
]
`;
